engine.py

```diff
@@ -112,18 +112,22 @@ def get_knight_moves(self, r, c, moves):
 
 	def get_king_moves(self, r, c, moves):
  		##TODO
+		rows = (r-1, r, r+1)
+		columns = (c+1, c, c-1)
 		if self.light_to_move: # if it's light's turn to move
+			box = ("  ", "d")
 			for i in range(len(self.board)):
 				for j in range(len(self.board[i])):
 					#if square in any direction is empty or has an opponent, move to square or move to capture
-					if (i == r-1 or i == r or i == r+1) and ((j == c+1) or (j == c-1) or j == c) and ((self.board[i][j] == "  ") or (self.board[i][j][1] == "d")):
+					if (i in rows) and (j in columns) and (self.board[i][j][1] in box or self.board[i][j] in box):
```

Comment:
if you effect the changes I suggested on line 118, this line should be

if (i in rows) and (j in columns) and (self.board[i][j][1] in available_squares)